### ISSUE TITLE
Add a sock argument to create_datagram_endpoint()

### ISSUE THREAD

**Issue by gvanrossum** [python/asyncio#266]
A number of requests for customization of datagram endpoints would be better served by having a sock argument to create_datagram_endpoint(), similar to create_server() and create_connection(). Example:

http://bugs.python.org/issue25230
http://bugs.python.org/issue23972

**Comment by gvanrossum** [comment]
See PR #267

**Comment by basak** [comment]
Is this intended to cover listening as a server as well as connecting as a client? I'm not clear on how to use this to get a socket listening.

**Comment by basak** [comment]
(and to test the PR I need both ends)

**Comment by gvanrossum** [comment]
You don't connect or listen when using UDP (datagrams); those are TCP (streams) terms. You can send at any time using `sendto()` and whenever you receive something your protocol's `datagram_received()` is called. (There is a `connect()` for UDP sockets but it merely supplies a default address argument to `sendto()`.)

**Comment by gvanrossum** [comment]
Try looking at examples/udp_echo.py in the asyncio project.

**Comment by basak** [comment]
Ah, thanks. I was failing to combine my understanding of Unix domain socket streams with Internet datagrams. I didn't realise I had to bind the "client" end to something too.

I've refactored my code to bind each end to a local name manually (using the `bind()` method on a `socket.socket` object) and then using `sock` against `create_datagram_endpoint()` works with #267. My "trivial" use case of JSON-RPC over Unix datagram sockets is now working perfectly, thanks!

Though I have learnt the lesson that I need to bind each client to a path, it does still save serialization/deserialization that I'd have to do on a stream socket.

Many thanks for asyncio. It makes so many things trivial again.
